test/conv3d.hpp

```diff
@@ -58,3 +59,19 @@ struct conv3d_driver : conv_driver<T>
         this->add(this->out_layout, "out_layout", this->generate_data({"NCDHW"}));
     }
 };
+
+class Conv3dFloat : public testing::TestWithParam<std::vector<std::string>>
```

Comment:
why are all these class definitions exactly the same?

Reply:
They are ostensibly configuration definitions split by precision. Though in this instance their data is defined entirely by the test fetching function and as such have no defined members at this level. It's conceivable that they could be summarized into a single class.